txnsync/incoming.go

```diff
@@ -120,7 +121,7 @@ func (s *syncState) asyncIncomingMessageHandler(networkPeer interface{}, peer *P
 		}
 	}()
 
-	incomingMessage := incomingMessage{networkPeer: networkPeer, sequenceNumber: sequenceNumber, encodedSize: len(message), peer: peer}
+	incomingMessage := incomingMessage{networkPeer: networkPeer, sequenceNumber: sequenceNumber, encodedSize: len(message), peer: peer, timeReceived: s.clock.Since()}
```

Comment:
So it turns out that using `s.clock.Since()` might be problematic since it causes a race condition. Additionally, the clock gets zeroed out when a new round starts. Is there a way around this?

Reply:
yes, but it's not super-trivial:
look at `func (tsnc *transactionSyncNodeConnector) Handle(raw network.IncomingMessage)`.
you'll see that `raw.Received` has the timestamp at which the message was received by the network socket.

If you'll change the `txnsync.IncomingMessageHandler` to include this field, you will be able to use it to determine absolute time. For the emulator, make sure that in case `Received` is zero, you have delay of zero.

Reply:
Does this mean for all timestamps made for the bandwidth calculations will need to be changed to use `time.Now()` instead of the WallClock? I haven't found a way to convert times from `time.Now()` to `s.clock.Since()`